lib/msf/core/exploit/smb/mitm_dispatcher.rb

```diff
@@ -0,0 +1,321 @@
+module Msf
+class Exploit
+module SMB
+  # This class provides a Man-In-The-Middle packet Dispatcher.
+  class MitmDispatcher < RubySMB::Dispatcher::Base
```

Comment:
We may want to rename/namespace this a bit, as there will be different types of 'man in the middle' smb exploits in framework in the future I'd imagine, i.e. for ntlm mitm etc. Will have to sync up with the team and decide on the naming convention that we'd want to follow :+1:

Reply:
Thanks! Let me know if I can help.

Reply:
Maybe `ShadowSMB2MitmDispatcher`?